### ISSUE TITLE
Issue with "ReflectionClassLikeTrait::getConstants" method on PHP 8.0 & 8.1

### ISSUE THREAD

**Issue by aik099** [goaop/parser-reflection#138]
I'm getting the following error when attempting to use a library on PHP 8.0 and PHP 8.1:

```
PHP Fatal error:  Declaration of Go\ParserReflection\Traits\ReflectionClassLikeTrait::getConstants() must be compatible with ReflectionClass::getConstants(?int $filter = null) in .../vendor/goaop/parser-reflection/src/ReflectionClass.php on line 224
```

The corresponding fix was made in https://github.com/goaop/parser-reflection/pull/125 and released as 4.0.0-RC1, but it has a minimal requirement of PHP 8.2.

Because of this, I'm unable to use a library on PHP 8.0 and PHP 8.1.

**Comment by aik099** [comment]
Closing, because I understand now, that there must be a reason not to support PHP 8.0 and PHP 8.1 in 4.x version of this library.

**Comment by lisachenko** [comment]
Actually, all <8.4 versions should be supported, except unions. Check the
master branch

Вт, 17 дек. 2024 г. в 09:53, Alex ***@***.***>:

**Comment by aik099** [comment]
> Actually, all <8.4 versions should be supported, except unions. Check the master branch

The library can scan code, built for PHP <= 8.4, BUT it can't be installed on PHP 8.0 and PHP 8.1 due to `composer.json` restriction (`"php": ">=8.2"`):

https://github.com/goaop/parser-reflection/blob/9c9bee019c76398f396ed673b6d7602b92a74a55/composer.json#L25C5-L25C19

**Comment by lisachenko** [comment]
Do you have any reasons to keep 8.0/8.1? Maybe just update your PHP
version? Or release a new version of your package for new PHP versions?

Вт, 17 дек. 2024 г. в 10:00, Alex ***@***.***>:

**Comment by aik099** [comment]
> Do you have any reasons to keep 8.0/8.1?

Not really. I was just surprised to see these 2 versions excluded from support. There were probably reasons for such a decision, and I respect them.

> Maybe just update your PHP version? Or release a new version of your package for new PHP versions?

I'll see what works best for me. Thank you.
